keel-core/src/main/kotlin/com/netflix/spinnaker/keel/lifecycle/LifecycleEvent.kt

```diff
@@ -0,0 +1,28 @@
+package com.netflix.spinnaker.keel.lifecycle
+
+import java.time.Instant
+
```

Comment:
I'm specifically using `artifactRef` here (a string) instead of the actual artifact because I had serialization problems with storing the whole artifact. The config name was getting saved as null and this was causing the first event to be disjoint from the other events. I know that using the whole object is better in general, but I felt like this was okay here.

Reply:
Ya, it would require a db look up. That's the other way I could probably do it - have a foreign key to the actual artifact. What would you recommend?

Reply:
kept the ref because serializing artifacts is way harder than normal and we don't need any fields from it.